### PR TITLE
Change compiling flag for hydrostatic options

### PR DESCRIPTION
**Pull request by XiaqiongZhou-NOAA**
## Description

Remove compiling flags (-DUSE_COND and -DMOIST_CAPPA) for hydrostatic runs.
inline post is modified to output omega and HGT for hydrostatic runs correctly.

## Testing

A new RT is added. It was tested on Hera with intel and gnu.

## Dependencies
- Related to ufs-weather-model [PR#2255](https://github.com/ufs-community/ufs-weather-model/pull/2255)

- waiting on NOAA-GFDL/GFDL_atmos_cubed_sphere (dev/emc) PR#[337](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/337)

### CODE REVIEW

File: ccpp/CMakeLists.txt [hunk 1]
```diff
@@ -84,8 +84,10 @@ endif(CCPP_32BIT)
 
 #------------------------------------------------------------------------------
 # Add model-specific flags for C/C++/Fortran preprocessor
+if (NOT HYDRO)
 add_definitions(-DMOIST_CAPPA -DUSE_COND -DNEMS_GSM)
```
Comment (BrianCurtis-NOAA) [inline ccpp/CMakeLists.txt]: please indent as you're inside an if conditional.

Comment (BrianCurtis-NOAA) [reply ccpp/CMakeLists.txt]: thx

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: io/post_fv3.F90 [hunk 2]
```diff
@@ -4343,6 +4340,28 @@ subroutine set_postvars_fv3(wrt_int_state,grid_id,mype,mpicomp)
...
       end do
 
+      do l=lm,1,-1
+!$omp parallel do default(none) private(i,j) shared(l,jsta,jend,omga,wh,dpres,zint,alpint,q,t,spval,ista,iend)
+        do j=jsta,jend
+          do i=ista,iend
+            if(wh(i,j,l) /= spval) then
+              if (omga(i,j,l) == spval .and. dpres(i,j,l) /= spval .and. zint(i,j,l) /=spval)  &
+                  omga(i,j,l) = (-1.) * wh(i,j,l) * dpres(i,j,l)/zint(i,j,l)
+              zint(i,j,l) = zint(i,j,l) + zint(i,j,l+1)
```
Comment (DusanJovic-NOAA) [inline io/post_fv3.F90]: What will happen on this line if `zint(i,j,l) == spval`?

Comment (XiaqiongZhou-NOAA) [reply io/post_fv3.F90]: added the conditions to ensure zint(i,j,l) and zint(i,j,l+1) is not spval.
I will re-run all regressions tests now if there are no more comments.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by zach1221** [comment]
@XiaqiongZhou-NOAA the cubed-sphere PR [#337](
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/337) has been merged. Can you please update the submodule hash and revert the url changes in .gitmodules?

Hash: https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/commit/cc8afe64b9420dd7379e47cee5e3f30f4ac44b58

**Comment by XiaqiongZhou-NOAA** [comment]
updated

**Pull request merged**